server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1931,7 +1931,9 @@ public ClusterState execute(ClusterState currentState) {
                         final SnapshotId foundId = snapshotsIdsInRepository.get(snapshotOrPattern);
                         if (foundId == null) {
                             if (snapshotIds.stream().noneMatch(snapshotId -> snapshotId.getName().equals(snapshotOrPattern))) {
-                                throw new SnapshotMissingException(repositoryName, snapshotOrPattern);
+                                final var snapshotMissingException = new SnapshotMissingException(repositoryName, snapshotOrPattern);
+                                logger.debug(snapshotMissingException.getMessage());
```

Comment:
The alternative is to log this message in [CleanupSnapshotStep](https://github.com/elastic/elasticsearch/blob/dfaec0dbf0c8a0ba2912381334019725848f1f5a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CleanupSnapshotStep.java#L66-L67) and potentially at info level? But I think logging here has wider coverage. Happy to adjust if needed.

Reply:
Please also let me know if you feel this should be tested.

Reply:
Should it be at info level so that is visible by default?

Reply:
I considered to add this at INFO level but did not do it because this logging is more of a bandaid solution. I think the actual fix is to log the initial deleting message only when we know exactly what to delete like David [suggested](https://github.com/elastic/elasticsearch/pull/100479#pullrequestreview-1663870234). The bandaid solution still has its value though since we can backport to 7.x. I hope this makes sense.